File: ProjectTracker20cs/www/PTWebService/App_Code/ResourceData.cs
using System;
using System.Collections.Generic;

public class ResourceData
{
  private int _id;
  private string _name;
  private List<ResourceAssignmentData> _projects = new List<ResourceAssignmentData>();

  public int Id
  {
    get { return _id; }
    set { _id = value; }
  }

  public string Name
  {
    get { return _name; }
    set { _name = value; }
  }

  public void AddProject(ResourceAssignmentData project)
  {
    _projects.Add(project);
  }

  public ResourceAssignmentData[] ResourceAssignments
  {
    get
    {
      if (_projects.Count > 0)
        return _projects.ToArray();
      return null;
    }
    set
    {
      _projects = new List<ResourceAssignmentData>(value);
    }
  }
}

File: ProjectTrackervb/www/PTWebService/App_Code/ResourceData.vb
Option Strict On

Imports Microsoft.VisualBasic

Public Class ResourceData

  Private _id As Integer
  Private _name As String
  Private _projects As New Generic.List(Of ResourceAssignmentData)

  Public Property Id() As Integer
    Get
      Return _id
    End Get
    Set(ByVal value As Integer)
      _id = value
    End Set
  End Property

  Public Property Name() As String
    Get
      Return _name
    End Get
    Set(ByVal value As String)
      _name = value
    End Set
  End Property

  Public Sub AddProject(ByVal project As ResourceAssignmentData)
    _projects.Add(project)
  End Sub

  Public Property ResourceAssignments() As ResourceAssignmentData()
    Get
      If _projects.Count > 0 Then
        Return _projects.ToArray

      Else
        Return Nothing
      End If
    End Get
    Set(ByVal value As ResourceAssignmentData())
      _projects = New Generic.List(Of ResourceAssignmentData)(value)
    End Set
  End Property

End Class
